src/components/ContentPreview/ContentPreview.js

```diff
@@ -493,7 +502,17 @@ class ContentPreview extends PureComponent<Props, State> {
      * @return {void}
      */
     fetchFileSuccessCallback = (file: BoxItem): void => {
-        this.setState({ file });
+        this.setState({ file, fileError: false });
+    };
+
+    /**
+     * File fetch success callback
+     *
+     * @param {Object} file - Box file
+     * @return {void}
+     */
+    fetchFileErrorCallback = (): void => {
+        this.setState({ fileError: true });
```

Comment:
Question, do we need to set the fileError state? Can we just try to reload the preview directly from the error callback? Setting the state will cause a re-render for react, and it seems that this variable is being used for BL

Reply:
We also should add the retry logic in here like its in preview library. It tries to load it 5 times, and then dies.

Reply:
You can maintain it in `this` and not necessarily need a state variable.
this.retryCount = 0 on successful fetch.

In that way you can eventually remove the logic from preview library.

Reply:
We could do retry here, but we don't set up any loading UI / the progress bar until we call preview.show(), so we would be blocked on rendering anything. Eventually we should expose that logic so that the element can control that kind of behavior. 

In the interim I could set a class property instead of using state as you suggested to avoid the re-render.